internal/pkg/git/remote.go

```diff
@@ -247,16 +247,28 @@ func (r *RemoteRepository) fetch(ctx context.Context) error {
 }
 
 func (r *RemoteRepository) runGitCmd(ctx context.Context, args ...string) (cmdResult, error) {
+	var lastErr error
```

Comment:
Mal som problem s padajucim testo v CI, tak som vylepsil tento kod:
1. Su tam retries/backoff.
  - Ked nastane context cancelled/deadline exceeded chyba pocas retry, tak nechcem tuto chybu, ale predchadzajucu chybu, ktora sposobila, ze sme museli robit retry.

Reply:
Tj. vysledkom tychto zmien je, ze sa dozviem preco to nejde, namiesto toho, ze to iba timeoutovalo.